eating_cookies/eating_cookies.py

```diff
@@ -3,14 +3,36 @@
 import sys
 
 # The cache parameter is here for if you want to implement
-# a solution that is more efficient than the naive 
+# a solution that is more efficient than the naive
 # recursive solution
+
+
 def eating_cookies(n, cache=None):
-  pass
+    if n == 0:
+        return 1
+    elif n < 0:
+        return 0
+
+    # Number of cookies cookie monster can eat is:
+    # 0, 1, 2, 3
+    ways = 0
+    for i in range(1, 4):  # start at 1
+        if cache is not None:
+            #  use cache list
+            if cache[n - i] == 0:
```

Comment:
I think it might be better if both these if statements were combined...
```
if cache is not None and cache[n - 1] == 0
    cache[n - i] = eating_cookies(n - i, cache)
    ways += cache[n - 1]
```
(I think you may even be able to shorthand `ways += cache[n - i] = eating_cookies(n - i, cache)`, but I haven't tested this yet.)